sstables_loader.cc

```diff
@@ -280,6 +280,9 @@ future<> sstable_streamer::stream_sstables(const dht::partition_range& pr, std::
             fmt::join(sst_processed | boost::adaptors::transformed([] (auto sst) { return sst->get_filename(); }), ", "));
         nr_sst_current += sst_processed.size();
         co_await stream_sstable_mutations(ops_uuid, pr, std::move(sst_processed));
+        if (on_streamed) {
+            std::invoke(on_streamed, batch_sst_nr);
+        }
```

Comment:
Two observations (in a sense that I don't suggest to do anything about it, just to draw your attention) about this place.

1. The progress turns out to be rather coarse. Not bytes (as for backup task), not even individual sstables, but "sixteen-sstables" steps
2. The `batch_sst_nr == sst_processed.size()` here, and all the code above this place uses the rightmost part of that invariant

Reply:
yeah, i knew. i should have put this in the commit message.